### DISCUSSION TITLE
How to set charging amps to max

### DISCUSSION THREAD

**Discussion by Jonathanese** [alandtse/tesla#643] [Q&A]
I currently have dynamic charging management. Basically, I have 3 charging speeds. Max, Slow, and Trickle.

Generally, the Slow speed is about 5A, and the Trickle speed is 1A.

When I'm home, the slow threshold is 40%, and the trickle threshold is 70%
When I'm away, the slow threshold is 80%, and the trickle threshold is 90%

That way, I can maximize how much free charging I get at work, while keeping the battery from getting too low, and charging way more slowly at high SoC. This has done me pretty well so far. I'm able to get about 80%+ of my charging for free at work, and 20% at home, while the battery spends most of its time in the 60%-70% range.

The issue is when it's below either of these thresholds, I want it to "max out" its speed. Two problems with that:
- You can't even try to set a value beyond the maximum. Instead of just doing the highest available, the action itself just fails.
- You can't set the value to the maximum, because it is changing all the time. If I set it to 48A, then if I go home, the script starts failing because I'm limited to 24A.

Potentially as a feature, there should probably be a "Charging Amps Max" item. Or the maximum Level2 rate should always be available, and the API just tries to get as close as it can.

Or perhaps there should be a "current max amps" entity that I could pull.

But basically, you can't really automate a value whose max keeps changing dynamically.

**Comment by fhteagle** [comment]
I think I found what you are looking for @Jonathanese :

If you put the action in your automation into YAML mode, you should be able to do this: 

```
service: number.set_value
data:
  value: >-
    {{ state_attr("number.charging_amps","max") }}
target:
  entity_id: number.charging_amps
```

If your Tesla integration has a name prefix for entities, you will need to edit "number.charging_amps" to be "number. <prefix_here>_charging_amps" .

Also, do you actually get any energy into the battery trickling at 1A? In my investigations, below about 8A made the energy wasted keeping the car on (BMS, coolant pumps, telemetry all powered up to the tune of maybe 300W) totally not worth it. Yes, workplace charging is no cost to you, but wasting energy is still wasting energy. Or are you just trying to "show charging" to avoid some kind of idle penalty?

**Reply by Jonathanese** [reply]
Turns out, I do still gain. I think the 1A charging setting is 1A _net_ after other loads. I did bump it to 2A though.

But basically the idea is that I could have it be essentially stopped at a level like 90%, but still be gaining _something_ while plugged in. And then I'm not entirely sure what I would be drawing while plugged in that wouldn't also be drawing when the car is sitting there, given that I use sentry mode quite regularly, so most of those systems would still be up and running anyway. I also figure while it's plugged in, I bump the polling rate up.

But yeah I'll see if this works! Sorry for digging this up after a few months. It's been a super busy summer.
